pandas/tests/groupby/test_function.py

```diff
@@ -1398,6 +1398,22 @@ def test_quantile_array_multiple_levels():
     tm.assert_frame_equal(result, expected)
 
 
+def test_groupby_quantile_with_arraylike_q_and_int_columns():
```

Comment:
can you parameterize this function by the dataframe size, e.g. so we have a 'small' and a 'large' (what you have now) example. I don't udnerstand why this worked for small before, but was reported in the OP as such. likely you can simply parameterize on the exponent (e.g. 1 and 4 would be enough)

Reply:
Done. parameterized the test and added more test cases to cover "small" and "large" dataframes. 

Although, I was not able to reproduce the "ok cases" mentioned by the OP with master code. Code still fails even with small frames, as long as q is not a scalar and the columns are integers starting from 0